web/public/docs/development.md

```diff
@@ -87,82 +142,121 @@ The order of these blocks does not matter, so you can shuffle the blocks.
+       ```javascript
+       "ui": [
+           "option1: {id: 'option1', type: 'choose', options: ['cat', 'dog'], placeholder: 'cat'}",
+           "option2: {id: 'option2', type: 'number', placeholder: 3}"
+          ],
+        ```
+*   an array with keys and values. Here, you have to use `" "` for the keys and
+    the strings. Definitions can also be mixed.
+
+In the   example below, we use a string as above for `option1` and an array with keys and values for `option2`. Note how for `option2` each key and value is defined as an individual string.
+
+ ```json
+ "ui": [
+  {"option1": "{id: 'option1', type: 'choose', options: ['cat', 'dog'], placeholder: 'cat'}"},
+  {"option2": {"id": "option2",
+               "type": "number",
+               "placeholder": 3}}],
+  ```
+
+#### flags
+Defines an array of flags which will be used by ImJoy to control the behavior of the plugin. Currently, we support these `flags` for run-time control. These flags allow to specify how ImJoy instances are handled by the Interface and the Plugin engine. For more information we refer to the section **TODO**.
+
+* `single-instance` (**for python plugins only**). Python engine will only run a single plugin process even if
+  plugin is called from multiple BUT identical workspaces. In this case, the different ImJoy instances will share
+  the same plugin process.
+
+* `allow-detach` (**for python plugins only**). Allows the plugin process to detatch from the user interface. This means
+  the plugin will not be killed when the user interface is disconnected or closed. However, in order to reconnect to this process,
+  you  also need to add the `single-instance` flag.
+
+Example: to make a plugin which can run without the user interface in the background and to which you can attach set
+ `"flags": ["single-instance", "allow-detach"]`. The inteface willautomaticallyreconnect to this process when re-launched.
+Please note that if  multiple ImJoy instances attache to this plugin process, each will call the `setup()` function.
+This may cause conflicts, we therefore recommend to (1) keep the interface-related code in the `setup()`, e.g. `api.register()`;
+(2) move code that you only want to run once per process into the `__init__` function of the plugin class.
+
+
+#### icon
+Defines the icon used in the plugin menu. You can find different icons here https://material.io/tools/icons/ and used the specified name. Or, you can directly copy and paste Emoji, for example from [here](https://getemoji.com/).
+
+#### inputs
+Defines the inputs with json-schema syntax (http://json-schema.org/) .
 For example, to define that the plugin uses png files, you can specify `"inputs": {"properties": {"type": {"enum": ["image/png"], "required": true}}, "type": "object"}` . You can also use the simplified format which assumes the inputs is an object and use json schema to describe the properties: `"inputs": {"type": {"enum": ["image/png"], "required": true}}`.
```

Comment:
Please remove the `, "required": true` in two places here, because I am trying to standardize the json-schema format, thanks.

Reply:
ok. done. will show up in next commit.